src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -48,6 +58,59 @@ public HttpConnectionDispatcher(ConnectionManager manager, ILoggerFactory logger
             }
         }
 
+        private async Task<bool> AuthorizeAsync<TEndPoint>(HttpContext context) where TEndPoint : EndPoint
```

Comment:
I would make a package like Microsoft.AspNetCore.Authentication.Policy and I would offer a service
IPolicyAuthenticationProvider {
    Task<ClaimsPrincipal> AuthenticateAsync(HttpContext context, string policyName);
}

It's implementation will do what the AuthorizeFilter does before setting the principal into the context. This has two benefits:
* You can reuse this logic across implementations (MVC, SignalR, any other middleware).
* You can get access to a principal that matches exactly what that policy defines at any point where you have access to the HttpContext.